Source/Core/DolphinQt2/Config/GameConfigWidget.cpp

```diff
@@ -0,0 +1,412 @@
+                                    const std::string& key)
+{
+  bool checked;
+
+  if (m_gameini_local.GetOrCreateSection(section)->Get(key, &checked))
+    return checkbox->setCheckState(checked ? Qt::Checked : Qt::Unchecked);
+  else if (m_gameini_default.GetOrCreateSection(section)->Get(key, &checked))
+    return checkbox->setCheckState(checked ? Qt::Checked : Qt::Unchecked);
+  else
+    checkbox->setCheckState(Qt::PartiallyChecked);
+}
+
+void GameConfigWidget::SaveCheckBox(QCheckBox* checkbox, const std::string& section,
+                                    const std::string& key)
+{
+  // Delete any existing entries from the local gameini if checkbox is undetermined.
+  // Otherwise, write the current value to the local gameini if the value differs from the default
+  // gameini values.
+  // Delete any existing entry from the local gameini if the value does not differ from the default
+  // gameini value.
+
+  if (checkbox->checkState() == Qt::PartiallyChecked)
+  {
+    m_gameini_local.DeleteKey(section, key);
+    return;
+  }
+
+  bool checked = checkbox->checkState() == Qt::Checked;
+
+  if (m_gameini_default.Exists(section, key))
+  {
+    bool default_value;
+    m_gameini_default.GetOrCreateSection(section)->Get(key, &default_value);
+
+    if (default_value != checked)
+      m_gameini_local.GetOrCreateSection(section)->Set(key, checked);
+    else
+      m_gameini_local.DeleteKey(section, key);
+
+    return;
+  }
+
+  m_gameini_local.GetOrCreateSection(section)->Set(key, checked);
+
+  return;
+}
+
+void GameConfigWidget::LoadSettings()
+{
+  // Load state information
+  m_state_combo->setCurrentIndex(m_game.GetRating());
+  m_state_comment_edit->setText(m_game.GetIssues());
+
+  // Load game-specific settings
+
+  // Core
+  LoadCheckBox(m_enable_dual_core, "Core", "CPUThread");
+  LoadCheckBox(m_enable_mmu, "Core", "MMU");
+  LoadCheckBox(m_skip_dcbz, "Core", "DCBZ");
+  LoadCheckBox(m_enable_fprf, "Core", "FPRF");
+  LoadCheckBox(m_sync_gpu, "Core", "SyncGPU");
+  LoadCheckBox(m_enable_fast_disc, "Core", "FastDiscSpeed");
+  LoadCheckBox(m_use_dsp_hle, "Core", "DSPHLE");
+
+  std::string determinism_mode;
+  if (!m_gameini_local.GetIfExists("Core", "GPUDeterminismMode", &determinism_mode))
+    if (!m_gameini_default.GetIfExists("Core", "GPUDeterminismMode", &determinism_mode))
+      determinism_mode = DETERMINISM_NOT_SET_STRING;
+
+  if (determinism_mode == DETERMINISM_AUTO_STRING)
+  {
+    m_deterministic_dual_core->setCurrentIndex(DETERMINISM_AUTO_INDEX);
+  }
+  else if (determinism_mode == DETERMINISM_NONE_STRING)
+  {
+    m_deterministic_dual_core->setCurrentIndex(DETERMINISM_NONE_INDEX);
+  }
+  else if (determinism_mode == DETERMINISM_FAKE_COMPLETION_STRING)
+  {
+    m_deterministic_dual_core->setCurrentIndex(DETERMINISM_FAKE_COMPLETION_INDEX);
+  }
+  else
+  {
+    m_deterministic_dual_core->setCurrentIndex(DETERMINISM_NOT_SET_INDEX);
+  }
+
+  // Stereoscopy
+  int depth_percentage = 100;
```

Comment:
There is quite a lot of code duplication here (there are 3 calls to `setValue` for one variable). I'd just do:

```
int depth_percentage = 100;
m_gameini_default.GetIfExists(...);
m_gameini_local.GetIfExists(...);
m_depth_slider->setValue(depth_percentage);
```